### PR TITLE
added logging interceptor

### PR DESCRIPTION
**Pull request by sashrika**

### CODE REVIEW

File: omod/src/main/java/org/openmrs/module/fhir/server/LogInterceptor.java [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+			if ("idOrResourceName".equals(theKey)) {
+				if (myRequestDetails.getId() != null) {
+					return myRequestDetails.getId().getValue();
+				}
+				if (myRequestDetails.getResourceName() != null) {
+					return myRequestDetails.getResourceName();
+				}
+				return "";
+			}
+			if (theKey.equals("requestParameters")) {
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/server/LogInterceptor.java]: equals should constant.equals
Please move strings to constants in all places

File: omod/src/main/java/org/openmrs/module/fhir/server/LogInterceptor.java [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+ * </tr>
+ * <tr>
+ * <td>${requestParameters}</td>
+ * <td>The HTTP request parameters (or "")</td>
+ * </tr>
+ * </table>
+ */
+public class LogInterceptor extends InterceptorAdapter {
+	
+	private static final org.slf4j.Logger ourLog = org.slf4j.LoggerFactory.getLogger(LogInterceptor.class);
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/server/LogInterceptor.java]: why you slfj here? can we use log4j?

File: omod/src/main/java/org/openmrs/module/fhir/server/LogInterceptor.java [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+	
+	private Logger myLogger = ourLog;
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/server/LogInterceptor.java]: why this need?

File: omod/src/main/java/org/openmrs/module/fhir/server/LogInterceptor.java [hunk 4]
```diff
@@ -0,0 +1,171 @@
...
+	
+	private String myMessageFormat = "${operationType} - ${idOrResourceName}";
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/server/LogInterceptor.java]: can make this final?

File: omod/src/main/java/org/openmrs/module/fhir/server/LogInterceptor.java [hunk 5]
```diff
@@ -0,0 +1,171 @@
...
+	public boolean incomingRequestPostProcessed(final RequestDetails theRequestDetails, final HttpServletRequest theRequest,
+	                                            HttpServletResponse theResponse) throws AuthenticationException {
+		
+		// Perform any string substitutions from the message format
+		StrLookup<?> lookup = new MyLookup(theRequest, theRequestDetails);
+		StrSubstitutor subs = new StrSubstitutor(lookup, "${", "}", '\\');
+		
+		// Actuall log the line
+		String line = subs.replace(myMessageFormat);
+		myLogger.info(line);
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/server/LogInterceptor.java]: why this is info? Lets make it debug

File: omod/src/main/java/org/openmrs/module/fhir/server/LogInterceptor.java [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+		private final RequestDetails myRequestDetails;
+		
+		private MyLookup(HttpServletRequest theRequest, RequestDetails theRequestDetails) {
+			myRequest = theRequest;
+			myRequestDetails = theRequestDetails;
+		}
+		
+		@Override
+		public String lookup(String theKey) {
+			if ("operationType".equals(theKey)) {
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/server/LogInterceptor.java]: move strings to constant

File: omod/src/main/java/org/openmrs/module/fhir/server/LogInterceptor.java [hunk 7]
```diff
@@ -0,0 +1,171 @@
...
+				return b.toString();
+			}
+			if (theKey.startsWith("requestHeader.")) {
+				String val = myRequest.getHeader(theKey.substring("requestHeader.".length()));
+				return StringUtils.defaultString(val);
+			}
+			if (theKey.startsWith("remoteAddr")) {
+				return StringUtils.defaultString(myRequest.getRemoteAddr());
+			}
+			return "!VAL!";
```
Comment (harsha89) [inline omod/src/main/java/org/openmrs/module/fhir/server/LogInterceptor.java]: special value?

**Comment by surangak** [comment]
Killed for #94

**Pull request closed without merging**